pkgs/python-packages/jmclient/default.nix

```diff
@@ -1,22 +1,63 @@
-{ version, src, lib, buildPythonPackage, fetchurl, future, configparser, joinmarketbase, joinmarketdaemon, mnemonic, argon2_cffi, bencoderpyx, pyaes, joinmarketbitcoin, klein, pyjwt, autobahn }:
+{
+  pipBuildHook,
+  version,
+  src,
+  lib,
+  buildPythonPackageWithDepsCheck,
+  argon2_cffi,
+  autobahn,
+  bencoderpyx,
+  configparser,
+  fetchurl,
+  future,
+  joinmarketbase,
+  joinmarketbitcoin,
+  joinmarketdaemon,
+  klein,
+  mnemonic,
+  pyjwt,
+  werkzeug
+}:
 
-buildPythonPackage rec {
+buildPythonPackageWithDepsCheck rec {
   pname = "joinmarketclient";
   inherit version src;
 
   postUnpack = "sourceRoot=$sourceRoot/jmclient";
 
-  checkInputs = [ joinmarketbitcoin joinmarketdaemon ];
-
-  propagatedBuildInputs = [ future configparser joinmarketbase mnemonic argon2_cffi bencoderpyx pyaes klein pyjwt autobahn ];
+  propagatedBuildInputs = [
+    argon2_cffi
+    autobahn
+    bencoderpyx
+    configparser
+    future
+    joinmarketbase
+    joinmarketbitcoin
+    joinmarketdaemon
+    klein
+    mnemonic
+    pyjwt
+    werkzeug
+  ];
 
   patchPhase = ''
     substituteInPlace setup.py \
       --replace "'klein==20.6.0'" "'klein>=20.6.0'"
     substituteInPlace setup.py \
-      --replace "'pyjwt==2.4.0'" "'pyjwt==2.6.0'"
+      --replace "'argon2_cffi==21.3.0'" "'argon2_cffi==23.1.0'"
+    substituteInPlace setup.py \
+      --replace "'pyjwt==2.4.0'" "'pyjwt==2.8.0'"
+    substituteInPlace setup.py \
+      --replace "'werkzeug==2.2.3'" "'werkzeug==2.3.7'"
   '';
 
+  # The unit tests can't be run in a Nix build environment
+  doCheck = false;
```

Comment:
If I set this to true the build keeps succeeding. Do you mean that the tests that do run are unnecessary?

Reply:
With `doCheck = true`, the logs show `Ran 0 tests in 0.000s`. Running the tests requires adding `pytestCheckHook`, which then shows that the tests hang. (Probably while setting up a bitcoind regtest instance. I haven't looked into this closely.)